pkg/controller/machine/machine_controller.go

```diff
@@ -122,7 +123,8 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	// Set the ownerRef with foreground deletion if there is a linked cluster.
-	if cluster != nil && len(m.OwnerReferences) == 0 {
+	if cluster != nil && shouldAdopt(m) {
+		klog.Infof("cluster %s/%s is adopting machine %s", cluster.Namespace, cluster.Name, m.Name)
```

Comment:
`Cluster` and `Machine` here should probably be capitalized to indicate they are referring to the resources and not the generic terms.